src/NuGet.Core/NuGet.Packaging/ArrayPool.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETFRAMEWORK || NETSTANDARD2_0
+
+using System.Collections.Generic;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Provides a resource pool that enables reusing instances of arrays.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Renting and returning buffers with an <see cref="ArrayPool{T}"/> can increase performance
+    /// in situations where arrays are created and destroyed frequently, resulting in significant
+    /// memory pressure on the garbage collector.
+    /// </para>
+    /// <para>
+    /// This class is thread-safe.  All members may be used by multiple threads concurrently.
+    /// </para>
+    /// </remarks>
+    internal class ArrayPool<T>
+    {
+        // .NET Framework/.NET Standard version of .NET Core's ArrayPool<T>, intentionally mimic'ing
+        // its API shape so that consumption of the code doesn't need to change between targets.
+
+        // Largest multiple of 4096 under default LOH threadshold
+        private const int MaxPooledArraySize = 81920;
+        private readonly SimplePool<T[]> _pool = new(() => new T[MaxPooledArraySize]);
+
+        /// <summary>
+        /// Retrieves a shared <see cref="ArrayPool{T}"/> instance.
+        /// </summary>
+        public static readonly ArrayPool<T> Shared = new();
```

Comment:
I understand that this class is internal, should we add a `private constructor` or mark it abstract like https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Buffers/ArrayPool.cs just to make sure an instance will not be created by mistake in the future?